src/Core/PostgreSQL/Connection.cpp

```diff
@@ -2,11 +2,46 @@
 
 #if USE_LIBPQXX
 #include <Common/logger_useful.h>
+#include <Common/CurrentMetrics.h>
+#include <Common/ProfileEvents.h>
 
 
+namespace CurrentMetrics
+{
+    extern const Metric PostgreSQLClientConnections;
+    extern const Metric PostgreSQLClientConnectionsIdle;
+    extern const Metric PostgreSQLClientConnectionsInUse;
+}
+
+namespace ProfileEvents
+{
+    extern const Event PostgreSQLClientConnectionsCreated;
+}
+
 namespace postgres
 {
 
+Connection::Lease::Lease(pqxx::connection & connection_) : connection(&connection_)
+{
+    CurrentMetrics::sub(CurrentMetrics::PostgreSQLClientConnectionsIdle, 1);
+    CurrentMetrics::add(CurrentMetrics::PostgreSQLClientConnectionsInUse, 1);
+}
+
+Connection::Lease::~Lease()
+{
+    CurrentMetrics::sub(CurrentMetrics::PostgreSQLClientConnectionsInUse, 1);
+    CurrentMetrics::add(CurrentMetrics::PostgreSQLClientConnectionsIdle, 1);
+}
+
+Connection::Lease::Lease(const Lease & other) : Lease(*other.connection) {}
```

Comment:
On copy we update metrics again for same connection, do we really need to do so? And `operator=` below does not. Do we really want to copy `Lease`, can't we make it noncopyable or move-only?